### PR TITLE
Check archived property on publish

### PR DESCRIPTION
**Pull request by james-owen**
_checkForArchived function in publish.js

### CODE REVIEW

**Comment by james-owen** [comment]
#613

File: lib/services/publish.test.js [hunk 1]
```diff
@@ -56,12 +56,32 @@ describe(_.startCase(filename), function () {
       rule1.returns(Promise.reject(new Error(0)));
       rule2.returns(Promise.resolve('foo.com/bar/baz'));
 
-      return fn([ () => Promise.reject(new Error(0)), rule2, rule3 ])
+      const uri = 'http://example.com/uri@published';
+
+      db.getMeta.returns(Promise.resolve({archived: false}));
```
Comment (manuelurenah) [inline lib/services/publish.test.js]: Please mind spacing
```suggestion
      db.getMeta.returns(Promise.resolve({ archived: false }));
```

File: lib/services/publish.test.js [hunk 2]
```diff
@@ -92,6 +112,8 @@ describe(_.startCase(filename), function () {
       modifyFn.returnsArg(0);
       db.get.returns(Promise.resolve(JSON.stringify(fakePage)));
       meta.getMeta.returns(Promise.resolve({}));
+      db.getMeta.returns(Promise.resolve({archived: false}));
```
Comment (manuelurenah) [inline lib/services/publish.test.js]: Same suggestion as before

File: lib/services/publish.test.js [hunk 3]
```diff
@@ -237,4 +265,32 @@ describe(_.startCase(filename), function () {
...
+
+    it('resolves with an empty error object if the meta.archived property is false', function () {
+      const uri = 'http://example.com/uri@published';
+
+      db.getMeta.returns(Promise.resolve({archived: false}));
+
+      return fn(uri)
+        .then(resp => {
+          expect(resp).to.eql({val:'', errors: {}});
+        });
```
Comment (manuelurenah) [inline lib/services/publish.test.js]: We could have this in one line
```js
.then(resp => expect(resp).to.eql({ val: '', errors: {} }));
```

File: lib/services/publish.test.js [hunk 4]
```diff
@@ -237,4 +265,32 @@ describe(_.startCase(filename), function () {
...
+    const fn = lib[this.title];
+
+    it('rejects with an error object if the meta.archived property is true', function () {
+      const uri = 'http://example.com/uri@published';
+
+      db.getMeta.returns(Promise.resolve({archived: true}));
+
+      return fn(uri)
+        .catch(err => {
+          const expectedErr = {val:'', errors: {_checkForArchived: 'The page is archived and cannot be published'}};
```
Comment (manuelurenah) [inline lib/services/publish.test.js]: Please mind spacing when using objects. I know this is being picky 😛 but it helps for code style and readability
```suggestion
          const expectedErr = { val:'', errors: { _checkForArchived: 'The page is archived and cannot be published' } };
```

File: lib/services/publish.js [hunk 5]
```diff
@@ -106,6 +106,22 @@ function _checkForDynamicPage(uri, { _dynamic }) {
   return bluebird.resolve(_dynamic);
 }
 
+/**
+ * @param {String} uri
+ * @returns {Promise|Undefined}
```
Comment (PedroAlbR) [inline lib/services/publish.js]: This method doesn't return `undefined`, we could remove it.

**Pull request merged**